src/rustc.rs

```diff
@@ -38,6 +38,33 @@ impl VersionMetaExt for VersionMeta {
     }
 }
 
+pub(crate) trait ChannelExt {
```

Comment:
Maybe this could be upstreamed?

Reply:
Not sure about that. `rustc_version` parses `stable` channel differently: https://github.com/Kimundi/rustc-version-rs/blob/1d692a965f4e48a8cb72e82cda953107c0d22f47/src/lib.rs#L185

Reply:
In that case I don't see a benefit in using `rustc_version::Channel` over just passing the string through to `cargo` in the container. Let's make `cargo` error on a wrong channel instead of duplicating that logic here.